### DISCUSSION TITLE
Tiled 1.8.0 Beta

### DISCUSSION THREAD

**Discussion by bjorn** [mapeditor/tiled#3226] [General]
This is tagged `v1.7.90`, but it is actually Tiled 1.8.0 Beta. Please provide feedback, either by commenting here, by opening issues or on the [Tiled Discord](https://discord.gg/39wbTv7) server. Thanks!

<hr /><em>This discussion was created from the release <a href='https://github.com/mapeditor/tiled/releases/tag/v1.7.90'>Tiled 1.8.0 beta</a>.</em>

**Comment by aganm** [comment]
Will the new added json extensions become the default extensions used in tiled over the xml ones?

**Reply by bjorn** [reply]
No, the XML formats remain the default for now. The additional extensions for JSON files are mainly added so that people can associate Tiled with the files based on their extension.

**Comment by hexus** [comment]
This is working _delightfully_ for me so far!

![image](https://user-images.githubusercontent.com/2626789/147834808-9d8edb75-053d-4371-bcea-878db33c547d.png)

Before, I had properties like these listed individually, with dot-separated names. 

For example, `components.friction` and `components.physics` as custom properties would end up as so in-game:

```json
{
  "components": {
    "friction": true,
    "physics": true
  }
}
```

The problems I had with this:

1. I regularly had to either remember or refer to code for the names and types of properties in my game
2. I couldn't comfortably use object types because:
   a. Not every object of the same "type" uses the same set of properties
   b. It would clutter the custom properties list for every object unnecessarily, e.g. if I only needed to set or change a few
   c. There would be a _lot_ of duplication if I tried making an object type for every possible component combination

This feature now completely solves this for me. It serves as a drop-in replacement because of the way I'm parsing map JSON, and the way these new custom properties are structured within said JSON.

Class property types allow me to define all available properties per "component" in my entity component system, and objects can then opt in for each component by adding a single property. 🎉 

Subtleties that I've noticed that give a great user experience:

1. I can keep open the property types editor, add new types, or add properties to types, and simultaneously update the properties of an object in the map to suit 😍 Very convenient
2. Adding a property to a class that's already used by an object has it immediately update in an object's properties menu

The only issues I've noticed so far are minor, and UI related only:

1. The "properties" menu of a selected world object scrolls up if I add a new property to any custom property type in the property types editor
2. If I collapse all custom properties for an object and then click any empty space in the map, they all expand again

Thanks so much for this! It's amazing! 🎉

**Comment by user72356** [comment]
Is the new regionEdited signal part of this release?

UPDATE: Just tried and it's not :-( When can we hope this feature to make it into a final release?

**Reply by bjorn** [reply]
That's still on a branch, but I'll see about including it and looking into #3193 before the final release.

**Reply by user72356** [reply]
> That's still on a branch, but I'll see about including it and looking into #3193 before the final release.

Cool, also please take a look at #3199 !
